### PR TITLE
ClickUP > Control Panel Tasks

### PR DESCRIPTION
**Pull request by Bani2791**
# 1. DID

## Advanced > IPQS

[DID - ConnexCS Documentation (bani-cp1--connexcs-docs.netlify.app)](https://bani-cp1--connexcs-docs.netlify.app/customer/did/#advanced)

**IP Quality Score (IPQS)**: It blocks spam calls based on CLIs. It won't allow the CLIs of having a specific spam score that you've selected to pass through the system. For more information on IPQS, [click here](https://www.ipqualityscore.com/).

 !!! Warning
    The **IPQS** feature requires a paid subscription. To enable this feature navigate to **Setup :material-menu-right: Settings :material-menu-right: Account** and enable this feature.
    The **cost** for a single lookup is **$0.0040**.

 You can set a **Max Daily Quantity** for your customer's lookups. This restricts them to using only the specific number you allocate, ensuring controlled usage

## Media > RTP Codec
[DID - ConnexCS Documentation (bani-cp1--connexcs-docs.netlify.app)](https://bani-cp1--connexcs-docs.netlify.app/customer/did/#media)

+ **RTP Codec**: This fields allows you to have more specific control over the Codecs you choose for your system. After the selection you can assign various **Permissions** to the Codecs you select.

    + **Types of Permissions include**:

        + **Except**: This permission allows you to block all the codecs apart from the ones in the Whitelist. Codecs that weren't included in the carrier's initial codec list won't be taken into consideration.

        +  **Offer**: Offer also blocks the codecs apart from the ones in the whitelist and provides flexibility to change the order of the codecs in the list as well. Thus, the first codec in the list is treated as the primary codec (at the output) even if it was the last codec in the list.

        +  **Consume**: Identical to mask but enables the transcoding engine even if no other transcoding related options are given.

        +  **Transcode**: Allows the addition of codecs in the offered codec list even if the codecs weren't included in the original list of codecs. Here, transcoding engine will be engaged meaning  behind-the-scenes translation process is happening to ensure communication.<br> You can only add those codecs which are supported by your device for both encoding and decoding process. <br>One limitation of using this option is that it will strip-off all the unsupported codecs. Note that using this option doesn't necessarily always engage the transcoding engine. If all codecs given in the transcode list were present in the original list of offered codecs, then no transcoding will be done.<br> When you use this permission it enables you to mark/modify the Ptime.

        +  **Strip**: This permission allows you to remove the selected codecs or RTP Payload types from the SDP. Codecs removed using this option behaves as if they never existed in the SDP.

  

        |Strip|Transcode|Explanation|

        |-----|---------|-----------|

        |G729A|Opus|G729A is unavailable for transcoding|

        +  **Mask**: This option allows you to filter-out the selected codec from the output. Mask works well in combination with **Transcode** option. For example,

        |Input/Offering side|Mask|Transcode|Output/Outgoing Offer|Explanation|

        |-------------------|-----|---------|--------------------|-----------|

        |G729A|G729A|Opus|Opus|Transcoding happens between G729A and Opus but output is Opus, G729A is filtered out from outgoing offer|

        |G729A||Opus|G729A and Opus| Transcoding happens between G729A and Opus but outputs are both Opus and G729A since G729A wasn't filtered out|

         + **Accept**: This option is similar to **Strip** and **Mask** but it isn't removed from the codecs offered list. When you select this option, the selected codec is offered to the remote peer (output/outgoing offer), if the remote peer rejects the offered (incoming) codec it will be used for transcoding and is accepted by the input/offering side.

        In short, Accept permission allows your device to use codecs offered by the remote peer even if they weren't your initial choice.

        |Input/Offering side|Accept|Transcode|Output/Outgoing Offer|Explanation|

        |-------------------|------|---------|---------------------|-----------|

        |G729A|G729A|Opus|Reject|Transcoding still happens between G729A and Opus|

    + **Ptime(ms)**: This value determines the length of time each box (RTP packet) can hold. A higher ptime means each packet carries a longer chunk of audio/video data (bigger box), while a lower ptime means shorter chunks (smaller boxes).

## Media > Flags
FLAGS: Enable this option if you want to remove STIR/SHAKEN information from an inbound call before forwarding it to the destination.

[DID - ConnexCS Documentation (bani-cp1--connexcs-docs.netlify.app)](https://bani-cp1--connexcs-docs.netlify.app/customer/did/#media)

# 2. Regex Generator 

## Customer >  IP Authentication
[Auth - ConnexCS Documentation (bani-cp1--connexcs-docs.netlify.app)](https://bani-cp1--connexcs-docs.netlify.app/customer/auth/#__tabbed_1_4)

Customer > IP Auth

3. **Current**: Enter the prefix for the destination number, or the CLI. You also have the option to generate a regular expression using AI Generate Regular Expression.

[Auth - ConnexCS Documentation (bani-cp1--connexcs-docs.netlify.app)](https://bani-cp1--connexcs-docs.netlify.app/customer/auth/#enable-sip-user-authentication)

## Customer > SIP User Auth

3. **Current**: Enter the prefix for the destination number, or the CLI. You also have the option to generate a regular expression using AI Generate Regular Expression.

# 3. New Flag Items

## Management > Customer >Tag> CLI

[Tags - ConnexCS Documentation (bani-cp1--connexcs-docs.netlify.app)](https://bani-cp1--connexcs-docs.netlify.app/tag/#create-a-cli-tag)

16.**Flags**: When you select **Forced** as **Yes** and select a **Database**, you get an option to select a flag which is **Performance CLI Selection**.

The Performance CLI Selection chooses the CLI with the best ASR.

+ *Stir Shaken Required*: Choose this option when no certificate is selected.

+ *Stir Shaken Replace*: Choose this option when you wish to apply your Stir-shaken certificate instead of which is  already applied (for a call).

## Management > Customer >Routing> CLI

[CLI - ConnexCS Documentation (bani-cp1--connexcs-docs.netlify.app)](https://bani-cp1--connexcs-docs.netlify.app/customer/cli/#creating-a-record)

# 4. AnyEdge 
[AnyEdge - ConnexCS Documentation (bani-cp1--connexcs-docs.netlify.app)](https://bani-cp1--connexcs-docs.netlify.app/anyedge/anyedge/#anyedge-destinations)

* **Weight**: You can allocate the server that permits the highest traffic or call volume. This means the server with the highest weight will handle more traffic or calls compared to other servers.

# 5. Carrier > Gateway Failure

[Carrier - ConnexCS Documentation (bani-cp1--connexcs-docs.netlify.app)](https://bani-cp1--connexcs-docs.netlify.app/carrier/#__tabbed_1_4)

* **Gateway Failover:** This feature allows you to swtich to another IP if the call(s) doesn't pass through or fails on the first IP. It has 4 options:
        * *None*: The call will not switch to any other IP.
        * *Failover 1*: The call will only to switch to 1 of the IP's that you've configured in the Auth .
        * *Failover 2*: The call will only to switch to 2 of the IP's that you've configured in the Auth .
        * *Failover 3*: The call will only to switch to 3 of the IP's that you've configured in the Auth section.

# 6. Customer > Routing
[Ingress Routing - ConnexCS Documentation (bani-cp1--connexcs-docs.netlify.app)](https://bani-cp1--connexcs-docs.netlify.app/customer/routing/#media)

**IP Quality Score (IPQS)**: It blocks spam calls based on CLIs. It won't allow the CLIs of having a specific spam score that you've selected to pass through the system. For more information on IPQS, [click here](https://www.ipqualityscore.com/).

 !!! Warning
    The **IPQS** feature requires a paid subscription. To enable this feature navigate to **Setup :material-menu-right: Settings :material-menu-right: Account** and enable this feature.
    The **cost** for a single lookup is **$0.0040**.

 You can set a **Max Daily Quantity** for your customer's lookups. This restricts them to using only the specific number you allocate, ensuring controlled usage.

!!! Note
    We don't charge you again if you repeat your lookup within 24-hours of time-span.

### CODE REVIEW

File: docs/carrier.md [hunk 1]
```diff
@@ -102,6 +102,11 @@ From the **Carriers** page, you can perform several management operations.
     * **Tags** and **TOML:** Not applicable to carriers.
     * **Ext.Accounting ID:** This is used to fill in work in an external accounting field to correlate between Connexcs and the external accountancy system.
     * **Call Recording Retention Days:** How long the customer wants to keep the recorded calls.
+    * **Gateway Failover:** This feature allows you to swtich to another IP if the call(s) doesn't pass through or fails on the first IP. It has 4 options:
+        * *None*: The call will not switch to any other IP.
+        * *Failover 1*: The call will only to switch to 1 of the IP's that you've configured in the Auth.
+        * *Failover 2*: The call will only to switch to 2 of the IP's that you've configured in the Auth.
+        * *Failover 3*: The call will only to switch to 3 of the IP's that you've configured in the Auth section.
```
Comment (AdamNawaz-connex) [inline docs/carrier.md:110]: Add something like this    Gateway Failover is a feature that enables automatic switching between multiple IP addresses configured within a carrier's authentication settings. By default, when a call reaches this carrier, it is attempted on one of the IPs configured in the carrier's authentication settings. If the call fails, it does not fail over to a different IP within the same carrier. However, enabling Gateway Failover allows the call to fail over to a different IP within the same carrier.

Options:
Gateway Failover typically offers several options that determine the behavior when a call fails on the first IP:

None: No failover occurs.
Failover 1: After a call fails on the first IP, the system attempts to route the call through one alternative IP configured in the authentication settings.
Failover 2: After a call fails on the first IP, the system attempts to route the call through up to two alternative IPs.
Failover 3: After a call fails on the first IP, the system attempts to route the call through up to three alternative IPs.

Comment (Bani2791) [reply docs/carrier.md:110]: done

File: docs/carrier.md [hunk 2]
```diff
@@ -102,6 +102,12 @@ From the **Carriers** page, you can perform several management operations.
     * **Tags** and **TOML:** Not applicable to carriers.
     * **Ext.Accounting ID:** This is used to fill in work in an external accounting field to correlate between Connexcs and the external accountancy system.
     * **Call Recording Retention Days:** How long the customer wants to keep the recorded calls.
+    * **Gateway Failover:**  enables automatic switching between multiple IP addresses configured within a carrier's authentication settings. By default, when a call reaches this carrier, it is attempted on one of the IPs configured in the carrier's authentication settings. If the call fails, it does not fail over to a different IP within the same carrier. However, enabling Gateway Failover allows the call to fail over to a different IP within the same carrier.
```
Comment (AdamNawaz-connex) [inline docs/carrier.md:105]: Enable   e capital

Comment (Bani2791) [reply docs/carrier.md]: done

**Review by AdamNawaz-connex — APPROVED** [review]

**Pull request merged**